src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/CreditManager.cs

```diff
@@ -155,5 +157,25 @@ public void Dispose()
                 }
             }
         }
+
+        private int TryRequestCreditNoLock(int amount)
+        {
+            if (_disposed)
```

Comment:
We should assert that the lock is held here.

Reply:
Done